### ISSUE TITLE
Getting different sentence embeddings when using model on CPU and GPU

### ISSUE THREAD

**Issue by kbkartik** [huggingface/transformers#18793]
### System Info

- `transformers` version: 4.21.2
- Platform: Linux-5.4.188+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.7.13
- Huggingface_hub version: 0.9.1
- PyTorch version (GPU?): 1.12.1+cu113 (False)
- Tensorflow version (GPU?): 2.8.2 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@LysandreJik @sg

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```from transformers import RobertaConfig, RobertaModel, RobertaTokenizer
import torch
import numpy as np
device = ("cuda" if torch.cuda.is_available() else "cpu")
# Initializing tokenizer
tokenizer = RobertaTokenizer.from_pretrained("roberta-base")

# Initializing a RoBERTa configuration
configuration = RobertaConfig()
configuration.vocab_size = tokenizer.vocab_size

# Initializing a model from the configuration
model = RobertaModel(configuration)
model = model.to(device)
model = model.eval()

with torch.no_grad():
    tokenized_task = tokenizer('random_sentence_check_v000', return_tensors="pt")
    outputs = model(**tokenized_task.to(device))
    embedding = outputs.pooler_output.squeeze(0).cpu().numpy().tolist()

### Expected behavior

I should get same sentence embedding either on CPU and GPU from a pretrained model

**Comment by kbkartik** [comment]
To fix this issue, we need to set the seed values which I forgot.
